solar/solar/cli/resource.py

```diff
@@ -63,33 +63,45 @@ def action(dry_run_mapping, dry_run, action, resource):
                 str(key)
             ))
 
+def backtrack_single(i):
+    def format_input(i):
+        return '{}::{}'.format(i.resource.name, i.name)
+
+    if isinstance(i, list):
+        return [backtrack_single(bi) for bi in i]
+
+    if isinstance(i, dict):
+        return {
+            k: backtrack_single(bi) for k, bi in i.items()
+        }
+
+    bi = i.backtrack_value_emitter(level=1)
+    if isinstance(i, orm.DBResourceInput) and isinstance(bi, orm.DBResourceInput) and i == bi:
+        return (format_input(i), )
+
+    return (format_input(i), backtrack_single(bi))
+
 @resource.command()
 @click.argument('resource')
 def backtrack_inputs(resource):
     r = sresource.load(resource)
 
-    inputs = []
-
-    def backtrack(i):
-        def format_input(i):
-            return '{}::{}'.format(i.resource.name, i.name)
-
-        if isinstance(i, list):
-            return [backtrack(bi) for bi in i]
-
-        if isinstance(i, dict):
-            return {
-                k: backtrack(bi) for k, bi in i.items()
-            }
+    for i in r.resource_inputs().values():
+        click.echo(yaml.safe_dump({i.name: backtrack_single(i)}, default_flow_style=False))
 
-        bi = i.backtrack_value_emitter(level=1)
-        if isinstance(i, orm.DBResourceInput) and isinstance(bi, orm.DBResourceInput) and i == bi:
-            return (format_input(i), )
 
-        return (format_input(i), backtrack(bi))
+@resource.command()
+@click.argument('resource')
+@click.argument('input_name')
+def effective_iv(resource, input_name):
```

Comment:
Changed, when https://github.com/Mirantis/solar/pull/280 will be merged, then shortcuts like this wouldn't make sense.